src/Main/main.cpp

```diff
@@ -37,14 +38,21 @@ int main(int argc, char** argv) {
       FOEDAG::Foedag::getGuiType(cmd->WithQt(), cmd->WithQml());
 
   FOEDAG::Compiler* compiler = nullptr;
-  if (cmd->CompilerName() == "openfpga")
-    compiler = new FOEDAG::CompilerOpenFPGA();
-  else
+  FOEDAG::CompilerOpenFPGA* opcompiler = nullptr;
+  if (cmd->CompilerName() == "openfpga") {
+    opcompiler = new FOEDAG::CompilerOpenFPGA();
+    compiler = opcompiler;
+    compiler->SetUseVerific(cmd->UseVerific());
+  } else {
     compiler = new FOEDAG::Compiler();
-  compiler->SetUseVerific(cmd->UseVerific());
+  }
 
   FOEDAG::Foedag* foedag = new FOEDAG::Foedag(
       cmd, mainWindowBuilder, registerAllFoedagCommands, compiler);
-
+  if (opcompiler) {
+    const std::string& binpath = foedag->Context()->BinaryPath().string();
+    opcompiler->setYosysExecPath(binpath + "/yosys");
```

Comment:
this code required yosis and vrt to be in the same folder as foedag. Why?

Reply:
Because that is the safe way to have the executable you really need. Users might have many versions of Yosys or VPR in the path /usr/bin or /usr/local/bin
This forces the user to create an install that uses the correct Yosys or VPR.
Create symlink of Yosys and VPR to the foedag bin folder.

In the private repo we build all tools and install them in the same directory.

Reply:
got it!